python/python/lancedb/embeddings/bedrock.py

```
@@ -15,6 +15,417 @@
+        embeddings = []
+        for i in range(0, len(images), self.batch_size):
+            j = min(i + self.batch_size, len(images))
+            batch = images[i:j]
+            embeddings.extend(self._parallel_get(batch))
+        return embeddings
+
+    def _parallel_get(self, images: Union[List[str], List[bytes]]) -> List[np.ndarray]:
+        """
+        Issue concurrent requests to retrieve image data
+        """
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Ruff (0.8.2)

Undefined name `concurrent`

(F821)